### PR TITLE
docs: Spelling mistakes

### PR DESCRIPTION
**Pull request by mcauser**

### CODE REVIEW

File: docs/library/ucollections.rst [hunk 1]
```diff
@@ -13,10 +13,10 @@ Classes
 .. function:: namedtuple(name, fields)
 
     This is factory function to create a new namedtuple type with a specific
-    name and set of fields. A namedtyple is a subclass of tuple which allows
+    name and set of fields. A namedtuple is a subclass of tuple which allows
     to access its fields not just by numeric index, but also with an attribute
     access syntax using symbolic field names. Fields is a sequence of strings
-    specifying field names. For compatibily with CPython it can also be a
+    specifying field names. For compatibly with CPython it can also be a
```
Comment (dhylands) [inline docs/library/ucollections.rst]: This should be compatability

File: docs/library/uctypes.rst [hunk 2]
```diff
@@ -11,7 +11,7 @@ module is to define data structure layout with about the same power as the
 C language allows, and the access it using familiar dot-syntax to reference
 sub-fields.
 
-.. seealso::
+.. see also::
```
Comment (dhylands) [inline docs/library/uctypes.rst]: I think that this should be left as seealso wihch is a spinx directive:
http://www.sphinx-doc.org/en/stable/markup/para.html#directive-seealso

**Comment by dhylands** [comment]
Just wanted say thanks for this.

File: docs/library/ucollections.rst (hunk 1 above)
Comment (mcauser) [inline docs/library/ucollections.rst]: Good catch, fixed.

File: docs/library/ucollections.rst [hunk 3]
```diff
@@ -13,10 +13,10 @@ Classes
 .. function:: namedtuple(name, fields)
 
     This is factory function to create a new namedtuple type with a specific
-    name and set of fields. A namedtyple is a subclass of tuple which allows
+    name and set of fields. A namedtuple is a subclass of tuple which allows
     to access its fields not just by numeric index, but also with an attribute
     access syntax using symbolic field names. Fields is a sequence of strings
-    specifying field names. For compatibily with CPython it can also be a
+    specifying field names. For compatability with CPython it can also be a
```
Comment (dxxb) [inline docs/library/ucollections.rst]: Should be _compatibility_ https://en.wiktionary.org/wiki/compatability

Comment (mcauser) [inline docs/library/ucollections.rst]: Right you are!

**Comment by pfalcon** [comment]
> Just wanted say thanks for this.

I can second this, but then there's also a general rule of thumb - please avoid submitting large pull requests. Based on the above comments, these proofreadings require proofreading themselves, and with 37 files blob, that will take effort to even *schedule*.

Please be sure to squash any changes made.

**Comment by mcauser** [comment]
No problem. The 37 files only have 1-6 characters changed in each. Seemed a little too small for separate pull requests.

Re: squash, is it better to `--amend` and `push -f`, or use the new GitHub PR squash and merge?
https://github.com/blog/2141-squash-your-commits

**Comment by dhylands** [comment]
I generally use ```--amend``` and ```push -f``` or in this case, I'd use ```rebase -i``` and ```push -f```.

github is branch based, so as long as the changes stay on the same branch, then you don't need to open a new PR.

**Comment by pfalcon** [comment]
+1 to what @dhylands said: preparing complete, ready-to-merge as-is patch is the best approach, or otherwise someone else will need to finish what you started (and other people have their own stuff to finish ;-) ). I known about and tried that github's "squash" feature, it may be helpful for lazy projects which were just drowning in the merge mess github offered before, but not good/flexible enough for projects which really care about clean history.

**Comment by pfalcon** [comment]
Don't forget that we have https://github.com/micropython/micropython/wiki/DevelWorkflow with some info.

File: docs/library/sys.rst [hunk 4]
```diff
@@ -9,7 +9,7 @@ Functions
 
 .. function:: exit(retval=0)
 
-   Terminate current program with a given exit code. Underlyingly, this
+   Terminate current program with a given exit code. Underlying, this
```
Comment (pfalcon) [inline docs/library/sys.rst]: This isn't a required change.

**Comment by pfalcon** [comment]
Ok, I've cleaned up/rebased this myself this time. Thanks for contribution!

**Pull request closed without merging**